cpp/tests/groupby/group_shift_test.cpp

```diff
@@ -0,0 +1,386 @@
+
+  size_type offset = -1;
+  auto slr         = cudf::make_default_constructed_scalar(column_view(val).type());
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, BackwardShiftWithoutNull_ValidScalar)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val{"a", "bb", "cc", "d", "eee", "f", "gg"};
+  cudf::test::strings_column_wrapper expected({"42", "42", "42", "42", "42", "42", "42"});
+
+  size_type offset = -4;
+  auto slr         = cudf::make_string_scalar("42");
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, BackwardShiftWithNull_ValidScalar)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val({"a", "bb", "cc", "d", "eee", "f", "gg"},
+                                         {1, 1, 0, 0, 1, 0, 1});
+  cudf::test::strings_column_wrapper expected({"", "gg", "42", "42", "eee", "42", "42"},
+                                              {0, 1, 1, 1, 1, 1, 1});
+
+  size_type offset = -2;
+  auto slr         = cudf::make_string_scalar("42");
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, ZeroShiftNullScalar)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val{"a", "bb", "cc", "d", "eee", "f", "gg"};
+  cudf::test::strings_column_wrapper expected({"a", "cc", "f", "gg", "bb", "d", "eee"});
+
+  size_type offset = 0;
+  auto slr         = cudf::make_default_constructed_scalar(column_view(val).type());
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, ZeroShiftValidScalar)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val{"a", "bb", "cc", "d", "eee", "f", "gg"};
+  cudf::test::strings_column_wrapper expected({"a", "cc", "f", "gg", "bb", "d", "eee"});
+
+  size_type offset = 0;
+  auto slr         = cudf::make_string_scalar("42");
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, VeryLargeForwardOffset)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val{"a", "bb", "cc", "d", "eee", "f", "gg"};
+  cudf::test::strings_column_wrapper expected({"42", "42", "42", "42", "42", "42", "42"});
+
+  size_type offset = 1024;
+  auto slr         = cudf::make_string_scalar("42");
+
+  test_groupby_shift_string(key, val, offset, *slr, expected);
+}
+
+TEST_F(groupby_shift_string_test, VeryLargeBackwardOffset)
+{
+  cudf::test::fixed_width_column_wrapper<K> key{1, 2, 1, 2, 2, 1, 1};
+  cudf::test::strings_column_wrapper val{"a", "bb", "cc", "d", "eee", "f", "gg"};
+  cudf::test::strings_column_wrapper expected({"42", "42", "42", "42", "42", "42", "42"});
```

Comment:
```suggestion
  strings_column_wrapper expected({"42", "42", "42", "42", "42", "42", "42"});
```

Already inside `cudf::test` namespace, so `cudf::test::` is not needed. (at all other places in this file).